### ISSUE TITLE
facebook pixel code is incorrect

### ISSUE THREAD

**Issue by nickolashe** [opencurrents/opencurrents#1018]
`  fbq('init', '146875852629962', {
    em: user_email,
    fn: user_firstname,
    ln: user_lastname,
  });`

should be user.email, user.first_name, user.last_name

**Comment by dannypernik** [comment]
@nickolashe thanks for the spot. I've pushed change to develop. Max won't be able to test this one so could you verify vars are correct and close? Thank you

**Comment by nickolashe** [comment]
@dannypernik sure, can you link reference the commit in this issue pls

**Comment by dannypernik** [comment]
https://github.com/opencurrents/opencurrents/blob/develop/openCurrents/templates/partials/head.html#L44

**Comment by nickolashe** [comment]
@dannypernik i don't think this code is working yet. you can verify that on any page by opening js console in dev tools. it needs to be django template vars instead, {{user.email}} e.g.

**Comment by MaxTattenbach** [comment]
@nickolashe @dannypernik it would be useful if you guys got together to look at both the pixel and the hubspot as well.  Hubspot seems to have an issue as well.

**Comment by dannypernik** [comment]
@nickolashe I've made the var change but not been able to see any difference it's made in JS console. I was initially getting a `BLOCKED_BY_CLIENT` error but discovered it was my ad blocker. When I turned off the blocker, the error disappeared from console and the FB Pixel Chrome extension identified the Pixel. However, the error disappears regardless of whether I use django vars or hard coded values, so this doesn't tell me whether the vars are being correctly captured or not. 

In fact, comparing my Pixel extension info to an image I found online suggests that the vars are not being passed.

Ours (see "Custom parameters sent"):
<img width="399" alt="screen shot 2018-01-25 at 12 31 43 pm" src="https://user-images.githubusercontent.com/3347095/35405514-00e48d8a-01cc-11e8-944a-71212d25ff7b.png">

Someone else's:
<img width="358" alt="screen shot 2018-01-25 at 12 31 24 pm" src="https://user-images.githubusercontent.com/3347095/35405533-109266f8-01cc-11e8-8fda-9a01fbdacac0.png">

You can get the chrome extension here: https://chrome.google.com/webstore/detail/facebook-pixel-helper/fdgfkebogiimcoedlicjlajpkdmockpc?hl=en

@MaxTattenbach I still have no idea what's up with Hubspot. Nicko can verify that Hubspot tracking is on every page (code is in `partials/end-scripts.html` and included at bottom of `layouts/base.htmk` and `layouts/home-base.html`) , and I've checked the devtools to confirm that tracking is firing properly on the profile page:

<img width="415" alt="screen shot 2018-01-25 at 12 58 17 pm" src="https://user-images.githubusercontent.com/3347095/35406813-8cc01d12-01cf-11e8-8233-24810166d339.png">
<img width="414" alt="screen shot 2018-01-25 at 12 58 34 pm" src="https://user-images.githubusercontent.com/3347095/35406814-8cd5628a-01cf-11e8-849a-a8ffd4522622.png">
<img width="414" alt="screen shot 2018-01-25 at 12 58 48 pm" src="https://user-images.githubusercontent.com/3347095/35406815-8ce87d70-01cf-11e8-9366-093fbe2f32f4.png">
<img width="360" alt="screen shot 2018-01-25 at 12 57 59 pm" src="https://user-images.githubusercontent.com/3347095/35406824-9205c254-01cf-11e8-9fd1-1c8bc561299e.png">

Max I'm following "How to verify that the tracking code is firing correctly" instructions midway through the page at https://knowledge.hubspot.com/articles/kcs_article/reports/how-do-i-know-if-my-hubspot-tracking-code-is-working

Feel free to confirm it yourself. I don't know what else to do without tech support.

**Comment by nickolashe** [comment]
@dannypernik i don't think i need to use any chrome extensions to see what's going on. you must be looking at the wrong place in chrome devtools if you are not seeing it. i'll post an image for each case separately below.

**Comment by nickolashe** [comment]
`    em: user.email,
    fn: user.first_name,
    ln: user.last_name`

produces:

<img width="1440" alt="screen shot 2018-01-25 at 1 48 58 pm" src="https://user-images.githubusercontent.com/6278359/35414160-8e766678-01d6-11e8-92a6-71a14e5fb980.png">

which can be further traced to:

<img width="1436" alt="screen shot 2018-01-25 at 1 49 58 pm" src="https://user-images.githubusercontent.com/6278359/35414217-b2ad3940-01d6-11e8-8d7a-648af9235750.png">

this is something we expect in this case since the var `user` is not defined in js, it exists only within django template context and must be accessed using the `{{ }}` syntax just like any other variable.

**Comment by nickolashe** [comment]
commit https://github.com/opencurrents/opencurrents/commit/3c79c5b1cb7707ea119bd9f5d7bd40a9a71f949e also produces a similar error (you should be able to verify that for yourself)

**Comment by nickolashe** [comment]
the reason why its failing is because the var gets expanded into a string, but in js all strings must be quoted, otherwise it will try to treat as a var and fail. it seems to be working when i quote all three vars, however it gives a warning about duplicate pixel id's now (again, see if you can reproduce this warning):

<img width="1439" alt="screen shot 2018-01-25 at 1 56 59 pm" src="https://user-images.githubusercontent.com/6278359/35414558-a9a6e318-01d7-11e8-814d-a46e9df90b88.png">

<img width="1438" alt="screen shot 2018-01-25 at 1 57 38 pm" src="https://user-images.githubusercontent.com/6278359/35414596-c15a6d54-01d7-11e8-9b62-51e3a893a1a7.png">

**Comment by nickolashe** [comment]
i'll let you run with this now that you know how to debug it. see if it solves the hubspot issue as well (do we have a separate issue for that one). please report your findings on both.

**Comment by dannypernik** [comment]
@nickolashe I've reproduced the previous errors and "Duplicate Pixel ID" error, though FB's Pixel Chrome extension says it found 1 Pixel.

As for Hubspot, I'm not clear how the Pixel issue is related. The following is the only code added for Hubspot and it does not include var names:
```
<script type=“text/javascript” id=“hs-script-loader” async defer src=“//js.hs-scripts.com/3045177.js”></script>
```

**Comment by nickolashe** [comment]
ok, lets discuss hubspot in a separate issue (do we have one yet)?

**Comment by nickolashe** [comment]
where are with pixel at this point - are we able to confirm that its doing what we want it to do?

**Comment by dannypernik** [comment]
I've attempted to send a form to the pixel from openc.herokuapp but have not received the name and email fields to the pixel dashboard. The pixel Chrome extension gives the following error:

<img width="406" alt="screen shot 2018-01-26 at 12 09 11 am" src="https://user-images.githubusercontent.com/3347095/35427125-38373e16-022d-11e8-9c92-f78c9c8cda0c.png">
<img width="398" alt="screen shot 2018-01-26 at 12 09 25 am" src="https://user-images.githubusercontent.com/3347095/35427126-38496082-022d-11e8-91d5-5fff920dc8b9.png">

The "Learn more" link navigates to the following page: https://developers.facebook.com/docs/facebook-pixel/pixel-helper#pii_invalid_parameter_value

**Comment by nickolashe** [comment]
I think this can mean that facebook is not finding this combination of first and last name?
Have you figured out why there is a duplicate warning?
Is there some tutorial you followed in setting this up?

**Comment by dannypernik** [comment]
Tried one suggestion on StackOverflow for fixing the duplicate Pixel error but to no avail. Here's the tutorial I followed:

<img width="641" alt="screen shot 2018-01-26 at 1 19 53 am" src="https://user-images.githubusercontent.com/3347095/35429065-030c8524-0238-11e8-8585-bc3050b9e9e2.png">
<img width="640" alt="screen shot 2018-01-26 at 1 20 07 am" src="https://user-images.githubusercontent.com/3347095/35429068-0509a348-0238-11e8-9aa9-3036a86b9feb.png">
<img width="619" alt="screen shot 2018-01-26 at 1 24 47 am" src="https://user-images.githubusercontent.com/3347095/35429071-07f538a6-0238-11e8-9484-ff8f428191be.png">
